EventGenerator/src/MuCapPhotonGenerator_tool.cc

```diff
@@ -0,0 +1,76 @@
+#include "art/Utilities/ToolMacros.h"
+
+#include "CLHEP/Random/RandPoissonQ.h"
+#include "CLHEP/Random/RandGeneral.h"
+
+#include "EventGenerator/inc/ParticleGeneratorTool.hh"
+
+#include "DataProducts/inc/PDGCode.hh"
+#include "MCDataProducts/inc/GenId.hh"
+#include "Mu2eUtilities/inc/RandomUnitSphere.hh"
+#include "Mu2eUtilities/inc/BinnedSpectrum.hh"
+#include "GlobalConstantsService/inc/GlobalConstantsHandle.hh"
+#include "GlobalConstantsService/inc/ParticleDataTable.hh"
+#include "GlobalConstantsService/inc/PhysicsParams.hh"
+
+namespace mu2e {
+  class MuCapPhotonGenerator : public ParticleGeneratorTool {
+  public:
+    explicit MuCapPhotonGenerator(Parameters const& conf) :// fhicl::ParameterSet const& pset)
+      _pdgId(PDGCode::gamma),
+      _mass(GlobalConstantsHandle<ParticleDataTable>()->particle(_pdgId).ref().mass().value()),
+      _genId(GenId::MuCapPhotonGenTool),
+      _rate(GlobalConstantsHandle<PhysicsParams>()->getCapturePhotonRate()),
+      _spectrumVariable(parseSpectrumVar(conf().physics().spectrumVariable())),
+      _spectrum(BinnedSpectrum(conf().physics()))
+    {
+
+    }
+      void generate(std::unique_ptr<GenParticleCollection>& out, const IO::StoppedParticleF& stop) override;
+
+    void setEngine(art::RandomNumberGenerator::base_engine_t& eng) {
+      _randomUnitSphere = new RandomUnitSphere(eng);
+      _randomPoissonQ = new CLHEP::RandPoissonQ(eng, _rate);
+      _randSpectrum = new CLHEP::RandGeneral(eng, _spectrum.getPDF(), _spectrum.getNbins());
+    }
+
+  private:
+    PDGCode::type _pdgId;
+    double _mass;
+    GenId _genId;
+    double _rate;
+
+    SpectrumVar       _spectrumVariable;
```

Comment:
MuCapPhotonGenererator does not need _mass and _spectrumVariable; please simplify.

Reply:
Done in commit ca938c135